internal/server.go

```diff
@@ -0,0 +1,44 @@
+package internal
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+)
+
+func StartServer(logger zerolog.Logger, filePath string, port int) error {
+
+	if filePath == "" {
+		return errors.New("no file path provided")
+	}
+
+	// set the server
+	server := newServer(logger)
+	err := server.setConn(filePath)
+	if err != nil {
+		return errors.New("error starting server database: " + fmt.Sprint(err))
+	}
+
+	// set the router
+	router := mux.NewRouter().StrictSlash(true)
```

Comment:
a problem is now that's there's no way to inject middlewares to the project, please consider having middlewares slice passed to NewServer